### PR TITLE
Upgrade to highcharts 6

### PR DESCRIPTION
**Pull request by anselmbradford**
Closes:
- [GHE]/CFGOV/platform/issues/2963
- [GHE]/CFGOV/platform/issues/3095
- [GHE]/CFGOV/platform/issues/3091
- [GHE]/CFGOV/platform/issues/3071
- [GHE]/CFGOV/platform/issues/3027
- [GHE]/CFGOV/platform/issues/3024
- [GHE]/CFGOV/platform/issues/2991
- https://github.com/cfpb/cfpb-chart-builder/issues/71

## Changes

- Fixes issue with TileMap where tiles at the edge of the map had their borders clipped.

- Fixes issue with TileMap where cursor and border highlight was not consistent when rolling over tile square and tile text. Sets cursor for both interactions to be default pointer and a consistent 2px border.

- Fixes an issue where the tooltip legend line that is 1px high disappeared sometimes (presumably because it wasn't landing on a full pixel value).

- Normalizes tooltip appearance to not have a drop-shadow on the text and to be of "square" style across all three line charts (line, line-comparison, and line-index) and GeoMap, and "callout" style in bar chart.

- Simplifies responsive styles to two states (except for TileMap, which has it's own responsive nuances that should be re-visited in the future). This fixes an issue where the range selector buttons had several breakpoints, yet still didn't have correct horizontal alignment at several screen sizes.

- Moves chart-type specific styles to their own stylesheets, to make it easier to see the scope of the styles.

- Adds additional code comments and organization of what styles control what.

- Slight tweaks to the layout based on assumptions of what looks best.

- Removes navigator scrollbar, which does not appear necessary since the navigator can be dragged and it does nothing at the default appearance.

## Testing

- Run `./setup.sh && gulp build && gulp watch`
- Visit http://localhost:5000/ and resize the page to see responsive sizes. Check in Chrome, FF, IE10+, Edge, iOS simulator.
- Interact with charts by hovering and checking applicable interactive bits.

### Test on cfgov-refresh
 - Pull down this branch.
 - Check out https://github.com/cfpb/cfgov-refresh/tree/upgrade_babel branch on cfgov-refresh
 - Run `./frontend.sh`
 - Replace `cfgov-refresh/node_modules/cfpb-chart-builder` with `cfpb-chart-builder` base directory from this branch.
 - Run `gulp clean && gulp build`
 - Visit CCT pages and mortgage performance trends on localhost.

## Screenshots

Before: 

 - [Desktop](https://user-images.githubusercontent.com/704760/47681564-3083e280-dba0-11e8-8cd9-554e930da7ac.png)

 - [Mobile](https://user-images.githubusercontent.com/704760/47681573-38438700-dba0-11e8-9752-dacdc234b921.png)

After:

 - [Desktop](https://user-images.githubusercontent.com/704760/47681584-3ed1fe80-dba0-11e8-81ff-f6badc432a10.png)

 - [Mobile](https://user-images.githubusercontent.com/704760/47682109-ca985a80-dba1-11e8-8550-559de0c01473.png)

## Notes

- ~~**GeoMaps:** I'm unclear on how the GeoMap is set up on the live site, since chart builder contains a large number of custom styles in `demo.less` for the demo GeoMap. The version on the master branch and in this PR differ in coloring, but I'm unclear if this is just something to do with the demo or if this will break the live maps. We may need to merge this, make a release, and then vigorously test with the live site, and release patches if needed before finally merging the bumped version into cfgov-refresh.~~ These have been removed

- ~~**Firefox** does not show a thicker border on roll over on the GeoMap, unlike Chrome, which does show such a border. This is an existing issue.~~ Fixed in https://github.com/cfpb/cfpb-chart-builder/pull/152/commits/73a209e0800f5618f3687cf21e5b4fbc9e76c83b

- **IE** does not support `transform` on SVG elements, which was needed for some fine tuning. Consequently the range selector text is not vertically aligned and the GeoMap zoom button text isn't centered. I couldn't figure out a way around this, and it's an improvement over what's live. The tooltip is buggy in IE11, which we need to confirm on the live maps with this update to see if it's an issue of the demo or the live maps. It's an existing issue in the master branch.

### CODE REVIEW

**Comment by anselmbradford** [comment]
I think we can ignore codecov in this case.

**Comment by contolini** [comment]
@anselmbradford Lemme double check the GeoMap anomaly.

File: package.json [hunk 1]
```diff
@@ -37,21 +37,21 @@
     "cf-core": "4.10.0",
     "cf-layout": "5.0.0",
     "cf-typography": "5.0.1",
-    "highcharts": "https://github.com/cfpb/highcharts-dist#fix-missing-plotline-label",
+    "highcharts": "6.2.0",
```
Comment (contolini) [inline package.json]: ☝️ This change is causing the geo map styling problems.

With the previous `#fix-missing-plotline-label` version of highcharts, the geo map looks like this:

screen shot 2018-11-01 at 6 53 34 pm

Inspecting unincorporated Montana shows that Highcharts applies an inline style to fill the svg path with the applicable light blue color:

`<path *shape and classes and other stuff* style="fill:#7eb7e8;vector-effect:non-scaling-stroke;"></path>`

With `6.2.0`, the geo map looks like this:

screen shot 2018-11-01 at 5 31 11 pm

The colors are off and inspecting unincorporated Montana shows:

`<path *path and classes and other stuff* style="stroke-width:inherit;"></path>`

The fork we were previously using was probably modified to fit our needs. Testing this PR with highcharts 5.0.14 (the latest version of 5.x) does not solve the problem. @cfarm do you remember why [we moved to the fork](https://github.com/cfpb/cfpb-chart-builder/commit/aeec92b488686d8432c5ff2f0f19de0a8b468849)?

**Comment by contolini** [comment]
After more testing it looks like the change happened between highcharts 5.0.12 and 5.0.13. Diffing those two tags shows a lot of changes to `dataClasses`. Reading the 5.0.13 changelog points to [this issue](https://github.com/highcharts/highcharts/issues/6988) that highlights the breaking change.

In styled mode, which we use, all colors are specified via CSS. Prior to 5.0.13, CSS could be overridden with JS styles. That changed with https://github.com/highcharts/highcharts/commit/a0febcb1bf35ee403c386ad7568099709266e944. So now when we assign the [color ramp values](https://github.com/cfpb/cfpb-chart-builder/blob/master/src/js/charts/GeoMap.js#L52) they are being ignored. We'll need to specify the colors via CSS or use 5.0.12.

**Comment by anselmbradford** [comment]
@contolini plz check now

screen shot 2018-11-02 at 3 04 56 pm

Note that I removed the demo styles, so green borders!

**Comment by contolini** [comment]
Nice fix! I haven't tested it against cf.gov but the demo page looks great!

**Comment by cfarm** [comment]
testing this out on cfgov-refresh

**Comment by anselmbradford** [comment]
Pushed up update to the logic for the y-axis values in the LineChart to fix:

screen shot 2018-11-06 at 2 59 43 pm

Now, if the values are greater than 10000 they will be converted to `M` format:

screen shot 2018-11-06 at 3 12 20 pm
screen shot 2018-11-06 at 3 12 35 pm

It's still tight, but not overlapping the label, so could maybe be adjusted in a future PR.

**Comment by cfarm** [comment]
Was able to test and confirm the charts work well across screen sizes in the following browsers/OS's: FF, Safari, Android, iOS iPhone 6, Edge.

I see issues in 2 browsers: IE10 and IE11. The charts do not load in cfgov-refresh, for example at this URL: http://localhost:8000/data-research/consumer-credit-trends/credit-cards/lending-borrower-age/

**Comment by anselmbradford** [comment]
> I see issues in 2 browsers: IE10 and IE11. The charts do not load in cfgov-refresh, for example at this URL: http://localhost:8000/data-research/consumer-credit-trends/credit-cards/lending-borrower-age/

The problem is we run the charts through core-js here to polyfill the things needed, so the demos in this repo work, however, when we go over to cfgov-refresh we don't use the compiled JS, we use the raw source and we don't polyfill it with core-js. Polyfilling it there (via the `usage` setting in webpack) leads to reaching into the `node_modules` of `cfpb-chart-builder` from `cfgov-refresh` and polyfilling its dependencies (e.g. highcharts), which in term runs into https://github.com/zloirock/core-js/issues/385 and IE falls over on errors of trying to set read-only properties when module.exports used in core-js code. Not totally certain how to handle it yet, perhaps it is as simple as adjusting [this regex](https://github.com/cfpb/cfgov-refresh/blob/master/config/webpack-config.js#L26) (which includes cf- and cfpb- packages) and making it so nested `node_modules` are not included.

**Comment by anselmbradford** [comment]
Additionally, IE10 started failing Travis on [this change](https://github.com/cfpb/cfpb-chart-builder/pull/152/commits/5e6c4ac70cb93f7ad9312150c21154a05dab17be), which I'm not clear on why.

**Comment by schaferjh** [comment]
Thanks for the heads up, @marteki 

@anselmbradford I screenshot or gif of this tooltip background behavior would be very helpful -- is that possible to provide? Thanks.

**Comment by anselmbradford** [comment]
@schaferjh I've fixed it on the linecharts, but it still affects the geomaps. It's also showing up in Edge and not just legacy IE unfortunately. Here's a gif https://github.com/highcharts/highcharts/issues/9374. I'm not sure if that's an acceptable tradeoff for now to fix the issues this closes, like https://github.cfpb.gov/CFGOV/platform/issues/3027

**Comment by schaferjh** [comment]
@anselmbradford Do you have a sense of how many visitors using the impacted browsers typically visit this section of the site?

**Comment by anselmbradford** [comment]
@schaferjh In the past year we've had 2,534 visitors, or 0.02% of total visitors, to https://www.consumerfinance.gov/data-research/mortgage-performance-trends/mortgages-90-or-more-days-delinquent/, of those 606 (23.91%) visited in IE and 112 (4.42%) visited in Edge. Meaning 28.33% of all visitors to that page are in IE/Edge (which is 0.007% of total visits to the site). IE11 is the top browser in that group, with almost 5-to-1 visits in IE11 vs any version of Edge.

Visits to /data-research/mortgage-performance-trends/mortgages-90-or-more-days-delinquent/ are slightly lower but roughly equivalent.

**Comment by schaferjh** [comment]
@anselmbradford ok, thanks for that. After chatting with @ascott1, we're comfortable moving forward with this PR and addressing the tooltip bug in a follow-on.

**Comment by anselmbradford** [comment]
Geomap issue is fixed! I'll merge this after Travis passes! Background in https://github.com/highcharts/highcharts/issues/9374#issuecomment-439175584

**Pull request merged**